openapi.yaml

```diff
@@ -1,190 +1,514 @@
+  filterAmenitiesSome:
+    in: query
+    name: filter[amenities][some]
+    description: Filter parks containing at least one of the listed amenities
+    required: false
+    type: array
+    collectionFormat: csv
+    items:
+      type: string
+      enum: *amenitiesEnum
+  filterName:
+    in: query
+    name: filter[name]
+    description: Filter parks by name
+    required: false
     type: string
-    required: true
-    description: '"Bearer [token]" where [token] is your OAuth2 access token'
-  pageNumber:
-    name: page[number]
+  filterCity:
     in: query
-    type: integer
+    name: filter[city]
+    description: Filter parks by city
     required: false
-    description: Page number of results
-    default: 1
-    minimum: 1
-  pageSize:
-    name: page[size]
+    type: string
+  filterState:
     in: query
-    type: integer
+    name: filter[state]
+    description: Filter parks by state (two letter abbreviation)
     required: false
-    description: Number of results to return
-    default: 25
-    maximum: 500
-    minimum: 1
+    type: string
+    pattern: &states '^(?:(A[KLRZ]|C[AOT]|D[CE]|FL|GA|HI|I[ADLN]|K[SY]|LA|M[ADEINOST]|N[CDEHJMVY]|O[HKR]|P[AR]|RI|S[CD]|T[NX]|UT|V[AIT]|W[AIVY]))$'
+  filterZip:
+    in: query
+    name: filter[zip]
+    description: Filter parks by zip code
+    required: false
+    type: integer
+  ownerIdPath:
+    in: path
+    name: ownerId
+    description: Unique Owner ID
+    type: string
+    required: true
+  parkIdPath:
+    name: parkId
+    in: path
+    description: Unique Park ID
+    required: true
+    type: string
+  parkPostBody:
+    name: parkBody
+    in: body
+    description: A park body object
+    required: true
+    schema:
+      $ref: '#/definitions/PostParkModel'
+  parkPatchBody:
+    name: parkBody
+    in: body
+    description: A park body object
+    required: true
+    schema:
+      $ref: '#/definitions/PatchParkModel'
+  ownerPostBody:
+    name: ownerPostBody
+    in: body
+    description: An owner body object
+    required: true
+    schema:
+      $ref: '#/definitions/PostOwnerModel'
+  ownerPatchBody:
+    name: ownerPatchBody
+    in: body
+    description: An owner body object
+    required: true
+    schema:
+      $ref: '#/definitions/PatchOwnerModel'
 definitions:
-  Meta:
-    properties:
-      totalResults:
-        type: integer
-        description: Total number of results
-        example: 10
-      totalPages:
-        type: integer
-        description: Total number of pages
-        example: 10
-      currentPageNumber:
-        type: integer
-        description: Page number of the returned results
-        example: 1
-      currentPageSize:
-        type: integer
-        description: Number of results per page
-        example: 25
   SelfLink:
     properties:
       self:
         type: string
+        example: request_url_here
         format: url
         description: Self-link of current resource
-  PaginationLinks:
```

Comment:
Is there any reason you removed pagination? It could be useful, since this API could return many parks at once as part of a broad query (all parks in Oregon, for example)

Reply:
Pagination would definitely be useful to make the API more scalable, but given the scope of the project, I think I'll leave it out for now.